src/vuestic-theme/vuestic-components/vuestic-collapse/VaCollapse.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="va-collapse">
+    <div
+      class="va-collapse__header"
+      :class="{
+        'vuestic-collapse__header--no-header': noHeader,
+      }"
+      @click="onHeaderClick()"
+    >
+      <template v-if="noHeader">
+        <slot name="header"/>
+      </template>
+      <div
+        v-else
+        class="va-collapse__header__content va-row "
+      >
+        <slot name="header"/>
+        <i
+          v-if="show"
+          class="fa fa-angle-up va-collapse__header__icon"
+        />
+        <i
+          v-else
+          class="fa fa-angle-down va-collapse__header__icon"
+        />
+      </div>
+    </div>
+    <div
+      class="va-collapse__body"
+      :class="{'va-collapse__body--with-background': isBackgroundExists}"
+      ref="collapseBody"
+    >
+      <slot name="body"/>
+    </div>
+
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'va-collapse',
+  props: {
+    value: {
+      type: Boolean
+    },
+    isBackgroundExists: Boolean
+  },
+  inject: {
+    accordion: {
+      default: () => ({
+        onChildChange: () => {
+        },
+      }),
+    },
+  },
+  data () {
+    return {
+      show: this.value,
+      noHeader: !this.$slots.header
+    }
+  },
+  watch: {
+    show (show) {
+      if (show) {
+        this.expand()
+      } else {
+        this.collapse()
+      }
+    },
+  },
+  methods: {
+    expand () {
+      const bodyContent = this.$refs.collapseBody
+      bodyContent.style.height = this.$slots.body[0].elm.clientHeight + 36 + 'px'
```

Comment:
Is that really needed?

Comment heavily if yes.

Reply:
Yes, that's really needed.